File: ts-tests/src/pages/iframe-page.ts
import { Page, FrameLocator } from '@playwright/test';

export class IframePage {
  readonly page: Page;

  constructor(page: Page) {
    this.page = page;
  }

  async navigate() {
    await Promise.all([
      this.page.goto('https://practice-automation.com/iframes/'),
      this.page.waitForSelector("iframe[name='top-iframe']")
    ]);
  }

  iframe(): FrameLocator {
    return this.page.frameLocator("iframe[name='top-iframe']");
  }
}

File: py-tests/pages/iframe_page.py
from playwright.sync_api import Page, FrameLocator


class IframePage:
    def __init__(self, page: Page) -> None:
        self.page = page

    def navigate(self) -> "IframePage":
        self.page.goto("https://practice-automation.com/iframes/")
        self.page.wait_for_selector("iframe[name='top-iframe']")
        return self

    def iframe(self) -> FrameLocator:
        return self.page.frame_locator("iframe[name='top-iframe']")
